File: tests/scala/src/main/scala/enum_/versions/Main_Conversions.scala
package enum_.versions

import dev.argon.verilization.scala_runtime.Converter

object Main_Conversions {
	val v3ToV4: Converter[Main.V3, Main.V4] = {
		case Main.V3.n(n) => Main.V4.n(n)
		case Main.V3.m(m) => Main.V4.m(m)
		case Main.V3.r(r) => Main.V4.r(Referenced.V4.fromV3.convert(r))
	}
}

File: tests/java/src/main/java/enum_/versions/Main_Conversions.java
package enum_.versions;

import dev.argon.verilization.runtime.Converter;

final class Main_Conversions {
    private Main_Conversions() {}
    
    static final Converter<Main.V3, Main.V4> v3ToV4 = new Converter<Main.V3, Main.V4>() {
        @Override
        public Main.V4 convert(Main.V3 prev) {
            return switch(prev) {
                case Main.V3.N prev2 -> new Main.V4.N(prev2.n());
                case Main.V3.M prev2 -> new Main.V4.M(prev2.m());
                case Main.V3.R prev2 -> new Main.V4.R(Referenced.V4.fromV3.convert(prev2.r()));
            };
        }
    };
    

}
